open-api.yaml

```diff
@@ -588,8 +588,7 @@ components:
         expiresAt:
           type:
             - string
-            - 'null'
```

Comment:
For me removing the `null` possibility is really painful for the users. And the current way of handling our keys does not solve the security issue. It means the users have to update the `expireAt` value -> the key is not changed and the security issue is still there (since the key does not change). It does not enforce the user at all to change the key, so to remove the security issue

For example, on GitHub side, with the API keys (called PAT on GitHub side) you can set an infinite expiration date. A warning is displayed, but I can
However, if I put an expiration date, I can never update the information once I created a key. Meaning I'm enforced to re-create a key if it expires, but I cannot change the expiration date.

Reply:
Removing the UPDATE request in the Meilisearch API will make the Integrations team's life easier. 
Still, on the other side, we will prevent the users from using keys more suitably for them like [**stripe**](https://stripe.com/docs/keys#limit-access) does, for example, they allow updates in particular fields. 

I could be wrong, but since we don't know exactly how the users will interact with our API's removing this capability seems to be not the best choice.

If we came up with a good solution for the key-rotation problem (see https://github.com/meilisearch/product/discussions/438), I think we can start thinking to remove the update method.

To sum up, I had a similar idea like @Kerollmops considered ([more here](https://github.com/meilisearch/product/discussions/403#discussioncomment-2503704)), and it seems to be the best in class because is able to solve the PATCH issue and to support the `null` value like Clem mentioned at the same time.

FYI: @gmourier